crates/provider-sdk/src/lib.rs

```diff
@@ -148,31 +150,125 @@ pub struct Context {
 /// The super trait containing all necessary traits for a provider
 pub trait Provider: ProviderHandler + WrpcNats + WrpcDispatch + Send + Sync + 'static {}
 
+/// Configuration of a link that is passed to a provider
+pub trait LinkConfig: Send + Sync {
+    /// Given that the link was established with the source as this provider,
+    /// get the target ID which should be a component
+    fn get_target_id(&self) -> &ComponentId;
+
+    /// Given that the link was established with the source as this provider,
```

Comment:
```suggestion
    /// Given that the link was established with the target as this provider,
```

Reply:
It's confusing naming, unfortunately. But folks should call this function when `put_target_config` is called